e2e/specs/age-based-vitals.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { type Visit } from '@openmrs/esm-framework';
+
+const createdPatients: Patient[] = [];
+const createdVisits: Visit[] = [];
+
+// Helper function to calculate birthdate from age
+function calculateBirthdate(age: { years?: number; months?: number }): string {
+  const date = new Date();
+  if (age.years) date.setFullYear(date.getFullYear() - age.years);
+  if (age.months) date.setMonth(date.getMonth() - age.months);
+  return date.toISOString().split('T')[0];
+}
+
+const ageGroups = [
+  {
+    name: 'newborn',
+    age: { months: 1 },
+    normalVitals: { temp: '36.7', systolic: '80', diastolic: '60', respiration: '40' },
+    criticalVitals: { temp: '37.8', systolic: '57', diastolic: '18', respiration: '18' },
+  },
+  {
+    name: 'infant',
+    age: { months: 6 },
+    normalVitals: { temp: '36.7', systolic: '80', diastolic: '50', respiration: '40' },
+    criticalVitals: { temp: '37.8', systolic: '65', diastolic: '37', respiration: '28' },
+  },
+  {
+    name: 'child',
+    age: { years: 8 },
+    normalVitals: { temp: '36.7', systolic: '100', diastolic: '70', respiration: '18' },
+    criticalVitals: { temp: '37.8', systolic: '85', diastolic: '55', respiration: '23' },
+  },
+  {
+    name: 'adult',
+    age: { years: 30 },
+    normalVitals: { temp: '36.7', systolic: '115', diastolic: '70', respiration: '16' },
+    criticalVitals: { temp: '37.8', systolic: '97', diastolic: '57', respiration: '11' },
+  },
+];
+
+test.describe('Vitals validation for different age groups', () => {
+  ageGroups.forEach((group) => {
+    test(`Normal vitals validation for ${group.name} patient`, async ({ api, page }) => {
+      const birthdate = calculateBirthdate(group.age);
+      const patient = await generateRandomPatient(api, { birthdate });
+      createdPatients.push(patient);
+      const visit = await startVisit(api, patient.uuid);
+      createdVisits.push(visit);
+
+      const vitalsPage = new BiometricsAndVitalsPage(page);
+      const headerRow = vitalsPage.vitalsTable().locator('thead > tr');
+      const dataRow = vitalsPage.vitalsTable().locator('tbody > tr');
+
+      await test.step(`When I visit the vitals and biometrics page for ${group.name} patient`, async () => {
+        await vitalsPage.goTo(patient.uuid);
+      });
+
+      await test.step('And I click the `Record biometrics` link to launch the form', async () => {
+        await vitalsPage.page.getByText(/record vital signs/i).click();
+      });
+
+      await test.step('Then I should see the `Record Vitals and Biometrics` form launch in the workspace', async () => {
+        await expect(vitalsPage.page.getByText(/record vitals and biometrics/i)).toBeVisible();
+      });
+
+      await test.step(`When I fill normal vitals: Temp=${group.normalVitals.temp}, BP=${group.normalVitals.systolic}/${group.normalVitals.diastolic}, Respiration=${group.normalVitals.respiration}`, async () => {
```

Comment:
In https://github.com/openmrs/openmrs-esm-patient-chart/pull/2479#discussion_r2156405882 by break downing steps, what I meant was to create a step per input. 

- When I enter the patient's temperature: ${group.normalVitals.temp}
- And I enter the systolic blood pressure: ${group.normalVitals.systolic}
- And I enter the diastolic blood pressure: ${group.normalVitals.diastolic}
- And I enter the respiration rate: ${group.normalVitals.respiration}